core-httpclient-impl/src/test/java/com/optimizely/ab/config/HttpProjectConfigManagerTest.java

```diff
@@ -128,53 +142,64 @@ public void testBuildDefer() throws Exception {
+        getResponse.setHeader(HttpHeaders.LAST_MODIFIED, modifiedStamp);
 
-        String datafile = handler.handleResponse(getResponse);
+        String datafile = projectConfigManager.getDatafileFromResponse(getResponse);
         assertNotNull(datafile);
 
         assertEquals("4", parseProjectConfig(datafile).getVersion());
+        // Confirm last modified time is set
+        assertEquals(modifiedStamp, projectConfigManager.getLastModified());
     }
 
     @Test(expected = ClientProtocolException.class)
-    public void testProjectConfigResponseHandler3XX() throws Exception {
-        ResponseHandler<String> handler = new ProjectConfigResponseHandler();
-
+    public void testGetDatafileHttpResponse3XX() throws Exception {
         HttpResponse getResponse = new BasicHttpResponse(new ProtocolVersion("TEST", 0, 0), 300, "TEST");
         getResponse.setEntity(new StringEntity(datafileString));
 
-        handler.handleResponse(getResponse);
+        projectConfigManager.getDatafileFromResponse(getResponse);
     }
 
-    @Test(expected = ClientProtocolException.class)
-    public void testProjectConfigResponseHandler4XX() throws Exception {
-        ResponseHandler<String> handler = new ProjectConfigResponseHandler();
+    @Test
+    public void testGetDatafileHttpResponse304() throws Exception {
+        HttpResponse getResponse = new BasicHttpResponse(new ProtocolVersion("TEST", 0, 0), 304, "TEST");
+        getResponse.setEntity(new StringEntity(datafileString));
 
+        String datafile = projectConfigManager.getDatafileFromResponse(getResponse);
+        assertNull(datafile);
+    }
+
+    @Test(expected = ClientProtocolException.class)
+    public void testGetDatafileHttpResponse4XX() throws Exception {
         HttpResponse getResponse = new BasicHttpResponse(new ProtocolVersion("TEST", 0, 0), 400, "TEST");
         getResponse.setEntity(new StringEntity(datafileString));
 
-        handler.handleResponse(getResponse);
+        projectConfigManager.getDatafileFromResponse(getResponse);
     }
 
     @Test(expected = ClientProtocolException.class)
-    public void testProjectConfigResponseHandler5XX() throws Exception {
-        ResponseHandler<String> handler = new ProjectConfigResponseHandler();
-
+    public void testGetDatafileHttpResponse5XX() throws Exception {
         HttpResponse getResponse = new BasicHttpResponse(new ProtocolVersion("TEST", 0, 0), 500, "TEST");
         getResponse.setEntity(new StringEntity(datafileString));
 
-        handler.handleResponse(getResponse);
+        projectConfigManager.getDatafileFromResponse(getResponse);
     }
 
-    @Test
     public void testInvalidPayload() throws Exception {
         reset(mockHttpClient);
-        when(mockHttpClient.execute(any(HttpGet.class), any(ResponseHandler.class)))
-            .thenReturn("I am an invalid response!");
+        CloseableHttpResponse invalidPayloadResponse = mock(CloseableHttpResponse.class);
```

Comment:
Can we use the `BasicHttpResponse` here?
```
new BasicHttpResponse(new ProtocolVersion("TEST", 0, 0), 200, "I am an invalid response!");
```

Reply:
`BasicHttpResponse` implements `HttpResponse` and not `CloseableHttpResponse` and so we can't use that here.